src/cc_helpers.cc

```diff
@@ -1611,6 +1611,35 @@ bool certifier::framework::cc_trust_manager::put_certifiers_in_store() {
                                  serialized_cert_messages);
 }
 
+bool certifier::framework::cc_trust_manager::write_private_key_to_file(
+    const string &filename) {
+  RSA *tmp_rsa = RSA_new();
+  if (!key_to_RSA(private_auth_key_, tmp_rsa)) {
+    printf(
+        "%s() error, line %d, Failed to convert private key to RSA format.\n",
+        __func__,
+        __LINE__);
+    return false;
+  }
+  FILE *outfh = fopen(filename.c_str(), "wb");
+  PEM_write_RSAPrivateKey(outfh,
```

Comment:
I am not sure if `PEM_write_RSAPrivateKey` handles null file pointers. You can add a check before calling to be on the safe side.